### PR TITLE
feat: add validation for deposit crypto step

### LINKED ISSUE #10: Validation: Implement validation of the exchange deposit crypto step for the CCXT adapter
**Description**

To enable and ensure smooth crypto deposits through the CCXT adapter, we need a solid validation process for the "exchange deposit crypto" step configuration. Before any deposit action is executed, this validation logic should review the configuration to ensure it matches expected standards, which will prevent any potential deposit errors. A typical configuration is below:

```
steps:
   - type: "exchange_deposit_crypto"
     adapter: "ccxt"
     config:
        exchange: "binance"
        currency: "USDC"
        amount: 123.45
        destination_address: "0x123456"
```

**Acceptance Criteria**

- [ ] The validation logic should check for the presence and validity of all mandatory fields (e.g., exchange, currency, amount, source_address).
- [ ] The validation process should identify and flag any inconsistencies or unsupported configurations.
- [ ] Proper error messages should be returned to guide users in case of configuration issues.
- [ ] Unit tests should be in place to ensure the validation logic catches typical mistakes and edge cases.

### PR DESCRIPTION
**Pull request by bayological**
### Description

Implemented validation for exchange deposit crypto step configuration.

### Other changes

- Added global variables to the configuration
- Added function to retrieve global variables to config loader

### Tested

- Added unit tests to verify the validation logic

### Related issues

-  Closes #10

### CODE REVIEW

File: packages/adapter-ccxt/src/types/ConfigCodecs.ts [hunk 1]
```diff
@@ -81,10 +81,35 @@ export const ExchangeSwap = t.type({
...
+// ---- Exchange.DepositCrypto ---- //
+
+/**
+ * Codec for the Exchange.ExchangeDepositCrypto configuration.
+ */
+export const ExchangeDepositCryptoConfigCodec = t.type({
+   exchange: t.string, // Identifier for the exchange
+   asset: t.string, // Asset to deposit e.g. "USDC"
+   toChain: t.string, // Chain identifier for destination chain
+   amount: t.number, // Amount to deposit TODO: This could be a value from the output from prev step
```
[the remark this answers is not in the document]
Comment (0xNghtly) [reply packages/adapter-ccxt/src/types/ConfigCodecs.ts:93]: Yeah it should be there and based on the conversation today with potential users it's likely this will not be implemented anytime soon. I will create an issue that covers the intention of this change and remove.

[the remark this answers is not in the document]
Comment (bayological) [reply packages/adapter-ccxt/src/types/ConfigCodecs.ts:93]: It still needs to be done but have created an issue and removed the todo:
https://github.com/mento-protocol/mate/issues/24

File: config.example.yaml [hunk 2]
```diff
@@ -33,15 +38,46 @@ flows:
...
-             fromAddress: 0xMyAddress1
+             fromAddress: 0xMyAddress1 -> TODO Remove and use primary address as default
              fromAmount: { stepOutput.ExchangeWithdrawCrypto.0.amount }
              toChain: 42220
              toToken: 0x765DE816845861e75A25fCA122bb6898B8B1282a
              toAddress: 0xMyAddress1
              maxSlippage: 0.01
+   - name: cUSD to USD
+     id: usd-to-cusd
+     description: "Bridge axlUSDC on Celo to USDC on Ethereum -> Deposit USDC to Binance Exchange -> Swap USDC to USD on Binance"
```
[the remark this answers is not in the document]
Comment (0xNghtly) [reply config.example.yaml:54]: Yep just an example but would help if the example was correct 😅

[the remark this answers is not in the document]
Comment (bayological) [reply config.example.yaml:54]: Yep, just an example but would help if the example was correct 😅
